credentials/src/main/java/me/uport/sdk/credentials/Credentials.kt

```diff
@@ -151,9 +154,113 @@ class Credentials(
         payload["claim"] = claim
         payload["vc"] = verifiedClaims ?: emptyList<String>()
         payload["callback"] = callbackUrl ?: ""
+
         return this.signJWT(payload, expiresInSeconds ?: 600L)
     }
 
+    /**
+     * Creates a W3C compliant [verifiable credential](https://w3c.github.io/vc-data-model/#credentials)
+     * serialized as a JWT.
+     *
+     * @param subject the subject of the claim in the [credential]
+     *      This becomes the `sub` field in the JWT
+     * @param credential the credential details; contains the claim being made about the [subject]
+     *      This becomes the `vc` field in the JWT
+     * @param notValidBefore [**optional**] the UNIX timestamp that marks the beginning of the validity period.
+     *      If this is not specified, the current timestamp is used as given by [clock]
+     *      This becomes the `nbf` field of the JWT. It is measured in seconds.
+     * @param validityPeriod [**optional**] the number of seconds of validity of this credential.
+     *      This influences the `exp` field in the JWT
+     *      If this is negative, there will be no expiry date set on the credential.
+     * @param audience [**optional**] the intended audience of this credential.
+     *      This results in the `aud` field of the JWT. If it is `null`, no audience will be set for the JWT.
+     * @param id [**optional**] the ID of this credential.
+     *      This becomes the `jti` field in the resulting JWT.
+     */
+    suspend fun createVerifiableCredential(
+        subject: String,
+        credential: CredentialParams,
+        notValidBefore: Long = clock.nowMs() / 1000L,
+        validityPeriod: Long = -1L,
+        audience: String? = null,
+        id: String? = null
+    ): String {
+
+        val payload = mutableMapOf<String, Any>()
+        payload["sub"] = subject
+
+        //add defaults if they are not set
+        val processedCredential = credential.copy(
+            context = credential.context.toMutableSet().apply { add("https://www.w3.org/2018/credentials/v1") }.toList(),
+            type = credential.type.toMutableSet().apply { add("VerifiableCredential") }.toList()
+        )
+
+        payload["vc"] = processedCredential
+        payload["nbf"] = notValidBefore
+        payload["iat"] = notValidBefore //for backward compatibility
+        if (validityPeriod >= 0) {
```

Comment:
I intentionally left `validityPeriod == 0` as a case to be considered.
Verifiers MAY use a broader interval when doing the check to account for clock skew so if a JWT is issued with 0 seconds of validity it will still be valid for the brief period of time that the validator accounts for.

Because of the way kotlin treats `null`-able types, it will throw a compile time error if the developer passes `null` as a validity period.